finetune/association.py

```diff
@@ -0,0 +1,353 @@
+import itertools
+import tensorflow as tf
+import numpy as np
+
+from finetune.base import BaseModel
+from finetune.target_encoders import SequenceLabelingEncoder
+from finetune.network_modules import association
+from finetune.crf import sequence_decode
+from finetune.utils import indico_to_finetune_sequence, finetune_to_indico_sequence
+from finetune.input_pipeline import BasePipeline, ENCODER
+from finetune.errors import FinetuneError
+from finetune.base import LOGGER
+
+class AssociationPipeline(BasePipeline):
+    def __init__(self, config, multi_label):
+        super(AssociationPipeline, self).__init__(config)
+        self.multi_label = multi_label
+        self.association_encoder = SequenceLabelingEncoder()
```

Comment:
Question -- should this be handled in a separate cls that inherits from SequenceLabelingEncoder and is returned by `_target_encoder`? Seems like it shouldn't live in the `__init__` of the pipeline.

Reply:
I agree